File: Interpreter C#/Interpreter/IfStmt.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Interpreter
{
    namespace Model
    {
        class IfStmt:IStmt
        {
            Exp exp;
            IStmt thenS;
            IStmt elseS;

            public IfStmt(Exp e, IStmt t,IStmt el)
            {
                this.exp = e;
                this.thenS = t;
                this.elseS = el;
            }

            public override string ToString()
   
            {
                return "IF(" + this.exp.ToString() + ") THEN(" + this.thenS.ToString() + ")ELSE(" + this.elseS.ToString() + ")";
            }

            PrgState IStmt.execute(PrgState state)
            {
                MyIStack<IStmt> exeStack = state.getExeStack();
                if (this.exp.eval(state.getSymTable())==0)
                {
                    exeStack.push(this.thenS);
                }
                else
                {
                    exeStack.push(this.elseS);
                }
                return state;
            }
        }
    }
}

File: MAP/Interpretor2/src/Model/IfStmt.java
package Model;

public class IfStmt implements IStmt {
	Exp exp;
	IStmt thenS;
	IStmt elseS;
	public IfStmt(Exp e,IStmt t,IStmt el)
	{
		this.exp=e;
		this.thenS=t;
		this.elseS=el;
	}
	@Override
	public String toString(){
		return "IF ("+exp.toString()+") THEN ("+thenS.toString()+") Else ("+elseS.toString()+")";
	}
	@Override
	public PrgState execute(PrgState state) {
		MyIStack<IStmt> exe=state.getExeStack();
		
		try {
			if(exp.eval(state.getSymTable(),state.getHeap())==0)
			{
				exe.push(elseS);
			}
			else
			{
				exe.push(thenS);
			}
		} catch (Exception e) {
			
		}
		return state;
	}

}
